### PR TITLE
feat(rds): support for local write forwarding in an Aurora Mysql cluster

### LINKED ISSUE #31523: rds: support for local write forwarding for an Aurora mysql cluster
### Describe the feature

To configure [local write forwarding](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-mysql-write-forwarding.html) feature for an Aurora mysql cluster.

### Use Case

Cloudformation supports enabling local write forwarding feature but AWS CDK does not support it.

### Proposed Solution

_No response_

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.159.0

### Environment details (OS name and version, etc.)

irrelevant

### PR DESCRIPTION
**Pull request by badmintoncryer**
### Issue # (if applicable)

Closes #31523 .

### Reason for this change

Cloudformation supports [enabling local write forwarding](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbcluster.html#cfn-rds-dbcluster-enablelocalwriteforwarding) feature but AWS CDK does not support it.

### Description of changes

- Add `enableLocalWriteForwarding` to `DatabaseClusterBaseProps`
- Add validation that `engineType` is either `aurora` or `aurora-mysql`
  - Having `engineType` set to `aurora` means launching a MySQL-compatible Aurora cluster.

### Description of how you validated changes

Add both unit and integ tests.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by go-to-k — CHANGES_REQUESTED** [review]
Thanks for the PR. I made one very minor comment.

(And below are just my note)

I ran CFn to see if CFn really throws when `enableLocalWriteForwarding` is false and engine type is not mysql. Then I got an error even though it was false. In other words, the validation was correct :)

```ts
    if (props.enableLocalWriteForwarding !== undefined && !['aurora', 'aurora-mysql'].includes(props.engine.engineType)) {
      throw new Error(`\'enableLocalWriteForwarding\' is only supported for Aurora Mysql cluster engine type, got: ${props.engine.engineType}`);
    }
```

File: packages/aws-cdk-lib/aws-rds/lib/cluster.ts [hunk 1]
```diff
@@ -385,6 +385,17 @@ interface DatabaseClusterBaseProps {
    * @default - false
    */
   readonly enableDataApi?: boolean;
+
+  /**
+   * Whether read replicas can forward write operations to the writer DB instance in the DB cluster.
+   *
+   * This setting can only be enabled for Aurora MySQL clusters.
```
[the remark this answers is not in the document]
Comment (badmintoncryer) [reply packages/aws-cdk-lib/aws-rds/lib/cluster.ts]: Added!

**Comment by badmintoncryer** [comment]
@go-to-k Thank you for your review! 

> Then I got an error even though it was false. 

This result was also something I did not anticipate. The validation using `!== undefined` was not intended to account for this but happened by chance.

I added a case to the unit test to verify that an error occurs even when false is passed!

**Comment by go-to-k** [comment]
It might be good to add a code comment that says that a CFn error will occur not only when the property is true but also when false, if the engine type is not mysql. Otherwise developers might get confused.

**Review by go-to-k — APPROVED** [review]
LGTM!

**Review by sumupitchayan — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**